jmclient/jmclient/blockchaininterface.py

```diff
@@ -215,6 +215,39 @@ def get_block(self, blockheight):
             return False
         return block
 
+    def rescanblockchain(self, start_height, end_height=None):
+        # Threading is not used in Joinmarket but due to blocking
+        # nature of this very slow RPC call, we need to fire and forget.
+        from threading import Thread
+        Thread(target=self.rescan_in_thread, args=(start_height,),
+               daemon=True).start()
+
+    def rescan_in_thread(self, start_height):
+        """ In order to not conflict with the existing main
+        JsonRPC connection in the main thread, this rescanning
+        thread creates a distinct JsonRPC object, just to make
+        this one RPC call `rescanblockchain <height>`, using the
+        same credentials.
+        """
+        from jmclient.jsonrpc import JsonRpc
+        authstr = self.jsonRpc.authstr
+        user, password = authstr.split(":")
+        newjsonRpc = JsonRpc(self.jsonRpc.host,
+                             self.jsonRpc.port,
+                             user, password,
+                             url=self.jsonRpc.url)
+        try:
+            newjsonRpc.call('rescanblockchain', [start_height])
+        except JsonRpcConnectionError:
+            log.error("Failure of RPC connection to Bitcoin Core. "
+                      "Rescanning process not started.")
+
+    def getwalletinfo(self):
```

Comment:
```suggestion
    def getwalletinfo(self) -> dict:
```

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1461/commits/7f4eaa9846f2f489480de1f2f8bfa9a1e2e96686